packages/electric-db-collection/src/electric.ts

```diff
@@ -669,6 +694,26 @@ function createElectricSync<T extends Row<unknown>>(
     sync: (params: Parameters<SyncConfig<T>[`sync`]>[0]) => {
       const { begin, write, commit, markReady, truncate, collection } = params
 
+      // Wrap markReady to wait for test hook in progressive mode
+      let progressiveReadyGate: Promise<void> | null = null
+      const wrappedMarkReady = () => {
+        // Only create gate if we're in buffering phase (first up-to-date)
+        if (
+          isBufferingInitialSync &&
```

Comment:
I don't like that we are closing over the `isBufferingInitialSync` which is actually define way down. I would rather explicitly pass it as an argument.